### PR TITLE
Voyage AI embeddings

### LINKED ISSUE #3146: Add Voyage AI embeddings [integration] [sync python integrations]
Sync Voyage AI embeddings python integration with the js library.

### PR DESCRIPTION
**Pull request by valdo99**
Adding VoyageEmbedding class to the embeddins.
Added integration tests.

the code is highly inspired by minmax and cohere :)

implemented the rest apis documented here: https://docs.voyageai.com/embeddings/

Fixes #3146

### CODE REVIEW

File: langchain/src/embeddings/voyage.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import { chunkArray } from "../util/chunk.js";
+import { getEnvironmentVariable } from "../util/env.js";
+import { Embeddings, EmbeddingsParams } from "./base.js";
+
+enum VoyageModelNames {
```
Comment (jacoblee93) [inline langchain/src/embeddings/voyage.ts:5]: Ah we can't use `enum` unfortunately as it'll break tree shaking :(

Can you use a type here instead?

File: langchain/src/embeddings/voyage.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+/**
+ * A class for generating embeddings using the Voyage AI API.
+ */
+export class VoyageEmbeddings
+  extends Embeddings
+  implements VoyageEmbeddingsParams
+{
+  modelName = VoyageModelNames.voyage01;
+
+  batchSize = 8;
```
Comment (jacoblee93) [inline langchain/src/embeddings/voyage.ts:51]: This seems a bit small?

File: langchain/src/embeddings/voyage.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+   * Constructor for the VoyageEmbeddings class.
+   * @param fields - An optional object with properties to configure the instance.
+   */
+  constructor(
+    fields?: Partial<VoyageEmbeddingsParams> & {
+      verbose?: boolean;
+      apiKey?: string;
+    }
+  ) {
+    const fieldsWithDefaults = { maxConcurrency: 2, ...fields };
```
Comment (jacoblee93) [inline langchain/src/embeddings/voyage.ts:71]: The default here was just for legacy purposes, you can leave it unset/set it higher if you choose

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Thanks for this! Mostly LGTM, only one important change.

File: langchain/src/embeddings/voyage.ts (hunk 2 above)
Comment (valdo99) [reply langchain/src/embeddings/voyage.ts:51]: https://docs.voyageai.com/embeddings/#via-http-request 

> as you can see: input - A single string or a list of strings. The maximum number of strings is 8.

File: langchain/src/embeddings/voyage.ts (hunk 3 above)
Comment (valdo99) [reply langchain/src/embeddings/voyage.ts:71]: i'll just leave it unset

File: langchain/src/embeddings/voyage.ts (hunk 1 above)
Comment (valdo99) [reply langchain/src/embeddings/voyage.ts]: @jacoblee93 i removed even the type, cause i see on the Voyage docs, that they have more embeddings model rolling out, so i'll just leave the possibility tu change the model name :)

**Comment by jacoblee93** [comment]
Thanks! You'll also want to add docs, and can you run `yarn build` from `langchain/` to update build artifacts?

**Comment by valdo99** [comment]
Yes my bad, added base docs + artifacts

**Comment by valdo99** [comment]
@jacoblee93 are we good now? :)

**Comment by jacoblee93** [comment]
LGTM thank you!

File: langchain/src/embeddings/voyage.ts (hunk 2 above)
Comment (oliviertassinari) [reply langchain/src/embeddings/voyage.ts:46]: The batch size logic used in Python: https://github.com/voyage-ai/langchain-voyageai/blob/225468c4e121745e14cd23e316d9633863206d0b/libs/voyageai/langchain_voyageai/embeddings.py#L61.

Comment (valdo99) [reply langchain/src/embeddings/voyage.ts:46]: i think it's better to create a new issue to improve based on the new logic :)
happy to help if this is needed fast :)

Comment (oliviertassinari) [reply langchain/src/embeddings/voyage.ts:46]: The current JavaScript logic seems like a good approximation of the Python one. No issues on my end.

**Pull request merged**